runtime/compiler/codegen/J9WatchedStaticFieldSnippet.cpp

```diff
@@ -0,0 +1,130 @@
+         J9Class *fieldClass;            // Declaring class of static field
+   } J9JITWatchedStaticFieldData;
+   */
+
+   // Emit each field and add a relocation record (for AOT compiles) for any field if needed.
+   J9JITWatchedStaticFieldData *str = reinterpret_cast<J9JITWatchedStaticFieldData *>(cursor);
+   str->method = staticFieldData.method;
+   str->location = staticFieldData.location;
+   str->fieldAddress = staticFieldData.fieldAddress;
+   str->fieldClass = staticFieldData.fieldClass;
+
+   if (cg()->comp()->getOption(TR_UseSymbolValidationManager))
+      {
+      cg()->addExternalRelocation(
+         new (cg()->trHeapMemory()) TR::ExternalRelocation(cursor + offsetof(J9JITWatchedStaticFieldData, method), reinterpret_cast<uint8_t *>(staticFieldData.method), reinterpret_cast<uint8_t *>(TR::SymbolType::typeMethod), TR_SymbolFromManager, cg()),
+         __FILE__,
+         __LINE__,
+         node);
+      }
+   else
+      {
+      cg()->addExternalRelocation(new (cg()->trHeapMemory()) TR::ExternalRelocation(cursor + offsetof(J9JITWatchedStaticFieldData, method), NULL, TR_RamMethod, cg()), __FILE__, __LINE__, node);
+      }
+
+   bool isResolved = !node->getSymbolReference()->isUnresolved();
+   // If the field is unresolved then we populate these snippet fields with the correct value at runtime (via the instructions generated by generateFillInDataBlockSequenceForUnresolvedField)
+   // and hence don't need to add relocation records here.
+   if (isResolved)
+      {
+      cg()->addExternalRelocation(
+         new (cg()->trHeapMemory()) TR::ExternalRelocation(cursor + offsetof(J9JITWatchedStaticFieldData, fieldAddress), reinterpret_cast<uint8_t *>(node->getSymbolReference()), reinterpret_cast<uint8_t *>(node->getInlinedSiteIndex()), TR_DataAddress, cg()),
+         __FILE__,
+         __LINE__,
+         node);
+
+      if (cg()->comp()->getOption(TR_UseSymbolValidationManager))
+         {
+         cg()->addExternalRelocation(
+            new (cg()->trHeapMemory()) TR::ExternalRelocation(cursor + offsetof(J9JITWatchedStaticFieldData, fieldClass), reinterpret_cast<uint8_t *>(staticFieldData.fieldClass), reinterpret_cast<uint8_t *>(TR::SymbolType::typeClass), TR_SymbolFromManager, cg()),
+            __FILE__,
+            __LINE__,
+            node);
+         }
+      else
+         {
+         // As things currently stand, this will not work on Power because TR_ClassAddress is used to a generate a 5 instruction sequence that materializes the address into a register. Meanwhile we are using TR_ClassAddress here to represent a contiguous word.
+         // A short-term solution would be to use TR_ClassPointer. However this is hacky because TR_ClassPointer expects an aconst node (so we would have to create a dummy node). The proper solution would be to implement the functionality in the power
+         // codegenerator to be able to patch TR_ClassAddress contiguous word.
```

Comment:
@AlenBadel FYI.. This is what I was referring to.

Reply:
Thinking on it, I believe even POWER can make `TR_ClassAddress` work; we would just need to add a new sequence kind (say `fixedSequence7`), and then in `TR_PPC64RelocationTarget::storeAddressSequence`, `fixedSequence7` means "no sequence", and you just have to patch the whole word. Then as long as power can load from a snippet (which would be a PC relative load), `TR_ClassAddress` would suffice.

Reply:
Thanks @dsouzai, I'm convinced that adding a new sequence kind to represent "no sequence" is necessary. I'll sure that capability is added to the Power codegen AOT.